### ISSUE TITLE
Issues playing videos with EIA 608 CC embeeded in the video

### ISSUE THREAD

**Issue by saiarcot895** [jasongdove/ErsatzTV#1235]
This seems to be a recent issue (within the last week or so), I'm not sure if it was caused by me upgrading to a development build or by some change in Plex.

On shows that have EIA 608 closed captioning embedded with the video, ErsatzTV might be unable to play those shows. They were playing fine till last week or so, and then I noticed videos starting to fail to be played. I'm not entirely sure what the cause for the change is; it's either upgrading ErsatzTV (from 0.7.4 to 0.7.5-f7d19e37) or upgrading Plex (from 1.31.0.6654-02189b09f to 1.31.3.6868-28fc46b27). The installation format for both are native packages/installations (not running in a container).

The error when playing is this:

```
System.InvalidOperationException: Sequence contains no matching element
   at System.Linq.ThrowHelper.ThrowNoMatchException()
   at ErsatzTV.Core.FFmpeg.FFmpegStreamSelector.SelectVideoStream(MediaVersion version) in /home/runner/work/ErsatzTV/ErsatzTV/ErsatzTV.Core/FFmpeg/FFmpegStreamSelector.cs:line 39
   at ErsatzTV.Core.FFmpeg.FFmpegLibraryProcessService.ForPlayoutItem(String ffmpegPath, String ffprobePath, Boolean saveReports, Channel channel, MediaVersion videoVersion, MediaItemAudioVersion audioVersion, String videoPath, String audioPath, Func`2 getSubtitles, String preferredAudioLanguage, String preferredAudioTitle, String preferredSubtitleLanguage, ChannelSubtitleMode subtitleMode, DateTimeOffset start, DateTimeOffset finish, DateTimeOffset now, Option`1 playoutItemWatermark, Option`1 globalWatermark, VaapiDriver vaapiDriver, String vaapiDevice, Option`1 qsvExtraHardwareFrames, Boolean hlsRealtime, FillerKind fillerKind, TimeSpan inPoint, TimeSpan outPoint, Int64 ptsOffset, Option`1 targetFramerate, Boolean disableWatermarks, Action`1 pipelineAction) in /home/runner/work/ErsatzTV/ErsatzTV/ErsatzTV.Core/FFmpeg/FFmpegLibraryProcessService.cs:line 294
   at ErsatzTV.Application.Streaming.GetPlayoutItemProcessByChannelNumberHandler.GetProcess(TvContext dbContext, GetPlayoutItemProcessByChannelNumber request, Channel channel, String ffmpegPath, String ffprobePath, CancellationToken cancellationToken) in /home/runner/work/ErsatzTV/ErsatzTV/ErsatzTV.Application/Streaming/Queries/GetPlayoutItemProcessByChannelNumberHandler.cs:line 272
   at ErsatzTV.Application.Streaming.FFmpegProcessHandler`1.Handle(T request, CancellationToken cancellationToken) in /home/runner/work/ErsatzTV/ErsatzTV/ErsatzTV.Application/Streaming/Queries/FFmpegProcessHandler.cs:line 24
   at ErsatzTV.Application.Streaming.FFmpegProcessHandler`1.Handle(T request, CancellationToken cancellationToken) in /home/runner/work/ErsatzTV/ErsatzTV/ErsatzTV.Application/Streaming/Queries/FFmpegProcessHandler.cs:line 24
   at ErsatzTV.Application.Streaming.HlsSessionWorker.Transcode(Boolean realtime, CancellationToken cancellationToken) in /home/runner/work/ErsatzTV/ErsatzTV/ErsatzTV.Application/Streaming/HlsSessionWorker.cs:line 363
```

This looks like it couldn't find any video stream for these files. I dug into the sqlite3 database a bit to see why it believed there was no video stream, and found this for one of the episodes:

```
sqlite> SELECT * FROM MediaStream WHERE MediaVersionId = 4172;
24497|0|eia_608||3|eng|0||0|0|4172|0||0||||||
24498|1|eac3||2|eng|6||1|0|4172|0||0||||||
24499|0|eia_608||3|eng|0||0|0|4172|0||0||||||
24500|2|srt||3|eng|0||0|0|4172|0||0||||||
```

This reads to be that it believed that stream 0 was just `eia_608` (without the `h264` video there), which then explains the error.

In my case, because I didn't really need the embedded closed captioning (and they were just annoying), I removed them with ffmpeg, and now it's working properly, but it would might be good to find out how that's happening.

**Comment by jasongdove** [comment]
This should be fixed with #1233 from earlier today. The regression was from when etv switched from using ffprobe with Plex libraries to trusting Plex to provide statistics. What happened is that Plex provides two streams both with index zero, and since etv expected the index to be unique, it overwrote the video stream with the subtitle stream info. I have fixed this by ignoring closed captions since they are otherwise unsupported by etv.

A similar issue that has also been fixed today has to do with external subtitles having no index number from Plex but deserializing to a default int of zero in etv.

Updating to the latest develop release should resolve these issues after the forced scan completes on startup.
